js/ui/appDisplay.js

```diff
@@ -594,9 +594,16 @@ const AllView = new Lang.Class({
                 this.addItem(icon);
         }));
 
-        // Add the App Center icon if it is enabled
-        if (global.settings.get_boolean(EOS_ENABLE_APP_CENTER_KEY))
-            this.addItem(new AppCenterIcon());
+        // Add the App Center icon if it is enabled and installed
+        if (global.settings.get_boolean(EOS_ENABLE_APP_CENTER_KEY)) {
+            let app = appSys.lookup_app(EOS_APP_CENTER_ID);
+            if (app) {
+                this.addItem(new AppCenterIcon(app));
+            } else {
```

Comment:
Updated. I don't like this style and found examples of my preferred style in this file ☺ but I'm happy to conform.